controller-server/src/test/java/com/yahoo/vespa/hosted/controller/certificate/EndpointCertificatesTest.java

```diff
@@ -215,12 +219,57 @@ public void includes_zones_in_deployment_spec_when_deploying_to_staging() {
                         "</deployment>\n");
 
         ZoneId testZone = tester.zoneRegistry().zones().all().in(Environment.staging).zones().stream().findFirst().orElseThrow().getId();
-        Optional<EndpointCertificateMetadata> endpointCertificateMetadata = endpointCertificates.getMetadata(testInstance, testZone, Optional.of(deploymentSpec.requireInstance("default")));
+        Optional<EndpointCertificateMetadata> endpointCertificateMetadata = endpointCertificates.getMetadata(testInstance, testZone, deploymentSpec);
         assertTrue(endpointCertificateMetadata.isPresent());
         assertTrue(endpointCertificateMetadata.get().keyName().matches("vespa.tls.default.default.*-key"));
         assertTrue(endpointCertificateMetadata.get().certName().matches("vespa.tls.default.default.*-cert"));
         assertEquals(0, endpointCertificateMetadata.get().version());
         assertEquals(Set.copyOf(expectedCombinedSans), Set.copyOf(endpointCertificateMetadata.get().requestedDnsSans()));
     }
 
+    @Test
+    public void includes_application_endpoint_when_declared() {
+        Instance instance = new Instance(ApplicationId.from("t1", "a1", "default"));
+        ZoneId zone1 = ZoneId.from(Environment.prod, RegionName.from("aws-us-east-1c"));
+        ZoneId zone2 = ZoneId.from(Environment.prod, RegionName.from("aws-us-west-2a"));
+        ApplicationPackage applicationPackage = new ApplicationPackageBuilder()
+                .instances("beta,main")
+                .region(zone1.region())
+                .region(zone2.region())
+                .applicationEndpoint("a", "qrs", zone2.region().value(),
+                                     Map.of(InstanceName.from("beta"), 2,
+                                            InstanceName.from("main"), 8))
+                .applicationEndpoint("b", "qrs", zone2.region().value(),
+                                     Map.of(InstanceName.from("beta"), 1,
+                                            InstanceName.from("main"), 1))
+                .applicationEndpoint("c", "qrs", zone1.region().value(),
+                                     Map.of(InstanceName.from("beta"), 4,
+                                            InstanceName.from("main"), 6))
+                .build();
+        ControllerTester tester = new ControllerTester(SystemName.Public);
+        EndpointCertificateValidatorImpl endpointCertificateValidator = new EndpointCertificateValidatorImpl(secretStore, clock);
+        EndpointCertificates endpointCertificates = new EndpointCertificates(tester.controller(), endpointCertificateMock, endpointCertificateValidator);
+        List<String> expectedSans = List.of(
```

Comment:
It is strange that this does not include san dns name for zone aws-us-west-2a ... But this is not really what you test here ...

Reply:
Agree, these tests should be rewritten to use DeploymentTester.